go/response_handlers.go

```diff
@@ -1761,3 +1681,114 @@ func handleSortedSetWithScoresResponse(response *C.struct_CommandResponse, rever
 
 	return zRangeResponseArray, nil
 }
+
+func handleXInfoStreamResponse(response *C.struct_CommandResponse) (models.XInfoStreamResponse, error) {
+	defer C.free_command_response(response)
+
+	typeErr := checkResponseType(response, C.Map, false)
+	if typeErr != nil {
+		return models.XInfoStreamResponse{}, typeErr
+	}
+
+	result, err := parseMap(response)
+	if err != nil {
+		return models.XInfoStreamResponse{}, err
+	}
+
+	infoMap, ok := result.(map[string]any)
+	if !ok {
+		return models.XInfoStreamResponse{},
+			fmt.Errorf("unexpected type of map: %T", result)
+	}
+
+	streamInfo := models.XInfoStreamResponse{}
+
+	// Parse integer fields
+	if val, ok := infoMap["length"].(int64); ok {
+		streamInfo.Length = val
+	}
+	if val, ok := infoMap["radix-tree-keys"].(int64); ok {
+		streamInfo.RadixTreeKeys = val
+	}
+	if val, ok := infoMap["radix-tree-nodes"].(int64); ok {
+		streamInfo.RadixTreeNodes = val
+	}
+	if val, ok := infoMap["groups"].(int64); ok {
+		streamInfo.Groups = val
+	}
+	if val, ok := infoMap["entries-added"].(int64); ok {
+		streamInfo.EntriesAdded = val
+	}
+
+	// Parse string fields
+	if val, ok := infoMap["last-generated-id"].(string); ok {
+		streamInfo.LastGeneratedID = val
+	}
+	if val, ok := infoMap["max-deleted-entry-id"].(string); ok {
+		streamInfo.MaxDeletedEntryID = val
+	}
+
+	// Get First Entry
+	entry := createEntry(infoMap, "first-entry")
+	if entry.ID != "" {
+		streamInfo.FirstEntry = entry
+	}
+
+	entry = createEntry(infoMap, "last-entry")
+	if entry.ID != "" {
+		streamInfo.LastEntry = entry
+	}
+
+	return streamInfo, nil
+}
+
+// Parse entry - it's an array where first element is ID and second is array of field-value pairs
+func createEntry(infoMap map[string]any, entryKey string) models.StreamEntry {
+	entry := models.StreamEntry{}
+	// Parse first-entry - it's an array where first element is ID and second is array of field-value pairs
+	if firstEntryArray, ok := infoMap[entryKey].([]any); ok && len(firstEntryArray) >= 2 {
+		// First element is the ID
+		if id, ok := firstEntryArray[0].(string); ok {
+			// Create field info array.
+			entryArray := []any{firstEntryArray[1]}
+			fieldInfos := createFieldInfoArray(entryArray)
+			// Create a StreamEntry with the ID
+			entry = models.StreamEntry{
+				ID:     id,
+				Fields: fieldInfos,
+			}
+		}
+	}
+	return entry
+}
+
+func createFieldInfoArray(entriesArray any) []models.FieldInfo {
+	fieldInfos := make([]models.FieldInfo, 0)
+	entriesData, ok := entriesArray.([]any)
+	if !ok {
+		entriesData = []any{}
+	}
+
+	for _, entryData := range entriesData {
+		fieldValuePairs, ok := entryData.([]any)
+		if !ok || len(fieldValuePairs) < 2 {
+			continue
+		}
+
+		if ok && len(fieldValuePairs) > 0 {
```

Comment:
This check is unnecessary given the check for `!ok || len(fieldValuePairs) < 2` evaluated to false at this point.